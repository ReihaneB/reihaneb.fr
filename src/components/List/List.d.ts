import React from 'react';

export interface ListProps {
  children: React.ReactNode;
  type?: 'ordered' | 'unordered';
}
