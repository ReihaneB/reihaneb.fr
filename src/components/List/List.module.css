.root {
  display: grid;
  gap: 16px;
  padding: revert;
}

ol,
ul {
  list-style: revert;
}
