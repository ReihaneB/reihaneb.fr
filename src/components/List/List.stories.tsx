import type { Meta, StoryObj } from '@storybook/react';

import List from './List';

const meta: Meta<typeof List> = {
  title: 'Components/List',
  component: List,
};

export default meta;
type Story = StoryObj<typeof List>;

const defaultArgs = {
  children: (
    <>
      <li>
        No one is perfect.
      </li>
      <li>
        But everyone can be better.
      </li>
    </>
  ),
};

export const Playground: Story = {
  args: {
    ...defaultArgs,
  },
};

export const Unordered: Story = {
  args: {
    ...defaultArgs,
    type: 'unordered',
  },
};

export const Ordered: Story = {
  args: {
    ...defaultArgs,
    type: 'ordered',
  },
};

export const LongOrdered: Story = {
  args: {
    ...defaultArgs,
    type: 'ordered',
    children: (
      <>
        <li>
          No one is perfect.
        </li>
        <li>
          But everyone can be better.
        </li>
        <li>
          It is not the strongest of the species that survive,
          nor the most intelligent, but the one most responsive to change.
        </li>
        <li>
          The only way to do great work is to love what you do.
        </li>
        <li>
          The best way to predict the future is to create it.
        </li>
      </>
    ),
  },
};

export const LongUnordered: Story = {
  args: {
    ...defaultArgs,
    children: (
      <>
        <li>
          No one is perfect.
        </li>
        <li>
          But everyone can be better.
        </li>
        <li>
          It is not the strongest of the species that survive,
          nor the most intelligent, but the one most responsive to change.
        </li>
        <li>
          The only way to do great work is to love what you do.
        </li>
        <li>
          The best way to predict the future is to create it.
        </li>
      </>
    ),
  },
};

export const Nested: Story = {
  args: {
    type: 'ordered',
    children: (
      <>
        <li>
          No one is perfect.
          <List type="unordered">
            <li>
              But everyone can be better.
            </li>
            <li>
              It is not the strongest of the species that survive,
              nor the most intelligent, but the one most responsive to change.
            </li>
            <li>
              The only way to do great work is to love what you do.
            </li>
            <li>
              The best way to predict the future is to create it.
            </li>
          </List>
        </li>
        <li>
          The only way to do great work is to love what you do.
          <List type="ordered">
            <li>
              The best way to predict the future is to create it.
            </li>
            <li>
              It is not the strongest of the species that survive,
              nor the most intelligent, but the one most responsive to change.
            </li>
            <li>
              But everyone can be better.
            </li>
            <li>
              No one is perfect.
            </li>
          </List>
        </li>
        <li>
          The best way to predict the future is to create it.
        </li>
      </>
    ),
  },
};
