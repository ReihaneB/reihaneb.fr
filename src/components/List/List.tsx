import { ListProps } from './List.d';
import styles from './List.module.css';

function List({
  children,
  type = 'unordered',
  ...rest
}: ListProps) {
  const Tag = type === 'ordered' ? 'ol' : 'ul';
  return (
    <Tag
      className={styles.root}
      {...rest}
    >
      {children}
    </Tag>
  );
}

export default List;
