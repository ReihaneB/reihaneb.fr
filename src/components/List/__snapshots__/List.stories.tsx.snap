// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/List LongOrdered smoke-test 1`] = `
<main class="-storybook-preview-module__root--fQckk inter-normal">
  <ol class="src-components-List-List-module__root--Z5ozz">
    <li>
      No one is perfect.
    </li>
    <li>
      But everyone can be better.
    </li>
    <li>
      It is not the strongest of the species that survive, nor the most intelligent, but the one most responsive to change.
    </li>
    <li>
      The only way to do great work is to love what you do.
    </li>
    <li>
      The best way to predict the future is to create it.
    </li>
  </ol>
</main>
`;

exports[`Components/List LongUnordered smoke-test 1`] = `
<main class="-storybook-preview-module__root--fQckk inter-normal">
  <ul class="src-components-List-List-module__root--Z5ozz">
    <li>
      No one is perfect.
    </li>
    <li>
      But everyone can be better.
    </li>
    <li>
      It is not the strongest of the species that survive, nor the most intelligent, but the one most responsive to change.
    </li>
    <li>
      The only way to do great work is to love what you do.
    </li>
    <li>
      The best way to predict the future is to create it.
    </li>
  </ul>
</main>
`;

exports[`Components/List Nested smoke-test 1`] = `
<main class="-storybook-preview-module__root--fQckk inter-normal">
  <ol class="src-components-List-List-module__root--Z5ozz">
    <li>
      No one is perfect.
      <ul class="src-components-List-List-module__root--Z5ozz">
        <li>
          But everyone can be better.
        </li>
        <li>
          It is not the strongest of the species that survive, nor the most intelligent, but the one most responsive to change.
        </li>
        <li>
          The only way to do great work is to love what you do.
        </li>
        <li>
          The best way to predict the future is to create it.
        </li>
      </ul>
    </li>
    <li>
      The only way to do great work is to love what you do.
      <ol class="src-components-List-List-module__root--Z5ozz">
        <li>
          The best way to predict the future is to create it.
        </li>
        <li>
          It is not the strongest of the species that survive, nor the most intelligent, but the one most responsive to change.
        </li>
        <li>
          But everyone can be better.
        </li>
        <li>
          No one is perfect.
        </li>
      </ol>
    </li>
    <li>
      The best way to predict the future is to create it.
    </li>
  </ol>
</main>
`;

exports[`Components/List Ordered smoke-test 1`] = `
<main class="-storybook-preview-module__root--fQckk inter-normal">
  <ol class="src-components-List-List-module__root--Z5ozz">
    <li>
      No one is perfect.
    </li>
    <li>
      But everyone can be better.
    </li>
  </ol>
</main>
`;

exports[`Components/List Playground smoke-test 1`] = `
<main class="-storybook-preview-module__root--fQckk inter-normal">
  <ul class="src-components-List-List-module__root--Z5ozz">
    <li>
      No one is perfect.
    </li>
    <li>
      But everyone can be better.
    </li>
  </ul>
</main>
`;

exports[`Components/List Unordered smoke-test 1`] = `
<main class="-storybook-preview-module__root--fQckk inter-normal">
  <ul class="src-components-List-List-module__root--Z5ozz">
    <li>
      No one is perfect.
    </li>
    <li>
      But everyone can be better.
    </li>
  </ul>
</main>
`;
